tools/hilow.wdl

```diff
@@ -0,0 +1,295 @@
+    ) + 2
+
+    command <<<
+        python /usr/local/bin/qc_hic.py \
+            --all_valid_pairs_stats ~{all_valid_pairs_stats} \
+            --mapping_stats_read1 ~{mapping_stats_read1} \
+            --mapping_stats_read2 ~{mapping_stats_read2} \
+            --pairing_stats ~{pairing_stats} \
+            ~{if defined(peaks_bed) then "--peaks_bed ~{peaks_bed}" else ""} \
+            ~{if defined(fithichip_bed) then "--fithichip_bed ~{fithichip_bed}" else ""} \
+            ~{(
+                if defined(fithichip_q01_bed)
+                then "--fithichip_q01_bed ~{fithichip_q01_bed}"
+                else ""
+            )} \
+            --prefix ~{prefix}
+    >>>
+
+    output {
+        File qc_report = prefix + "_QCreport.txt"
+    }
+
+    runtime {
+        container: "ghcr.io/stjudecloud/hilow:branch-hic_workflow-1.0.0"
+        cpu: 1
+        disks: "~{disk_size_gb} GB"
+        memory: "4 GB"
+        maxRetries: 1
+    }
+}
+
+task filter {
+    meta {
+        description: "Filter Hi-C results using an exclude list"
+        outputs: {
+            filtered_pairs: "Filtered pairs file",
+            removed_pairs: "Removed pairs file",
+            filtered_stats: "Filtered statistics file",
+        }
+    }
+
+    parameter_meta {
+        all_valid_pairs: "Valid pairs file"
+        chromsizes: "Chromosome sizes file"
+        exclude_list: "List of pairs to exclude"
+        padding: "Number of bases to add to each feature"
+    }
+
+    input {
+        File all_valid_pairs
+        File chromsizes
+        File exclude_list
+        Int padding = 50
+    }
+
+    String base = basename(exclude_list, ".gz")
+    String prefix = basename(all_valid_pairs, ".AllValidPairs")
+    Int disk_size_gb = ceil(size(all_valid_pairs, "GiB")) + 2
+
+    #@ except: LineWidth
+    command <<<
+        set -euo pipefail
+
+        gunzip -c ~{exclude_list} > ~{base} \
+           || ln -sf ~{exclude_list} ~{base}
+
+        #left bed
+        awk -F "\t" -v OFS="\t" '{print $2,$3,$3,$1}' ~{all_valid_pairs} \
+        | slopBed -i stdin -b ~{padding} -g ~{chromsizes} \
+        | intersectBed -a stdin -b ~{base} -u > left.bed
+
+        #right bed
+        awk -F "\t" -v OFS="\t" '{print $5,$6,$6,$1}' ~{all_valid_pairs} \
+        | slopBed -i stdin -b ~{padding} -g ~{chromsizes} \
+        | intersectBed -a stdin -b ~{base} -u > right.bed
+
+        cat <(cut -f 4 left.bed) <(cut -f 4 right.bed)|sort -u > filter.pair
+
+        python /usr/local/bin/filter_hic.py \
+            --all_valid_pairs ~{all_valid_pairs} \
+            --filter_pair filter.pair \
+            --prefix ~{prefix}
+
+        all=$(wc -l ~{all_valid_pairs} |cut -d " " -f 1)
+        filtered=$(wc -l ~{prefix}.allValidPairs.removed | cut -d " " -f 1)
+        percentage=$(echo "$filtered/$all*100"|bc -l)
+        echo -e "$filtered read pairs are filtered out from a total of $all, accounting for ${percentage} %\n" \
+            | tee \
```

Comment:
What's this `tee` doing?

Reply:
Good catch. It's supposed to write the information to the file and to STDOUT.